NIGHTLY_BUILD.md

```diff
@@ -0,0 +1,267 @@
+# Ice Nightly Builds
+
+Nightly builds of Ice are generated by GitHub Actions from the main development branch. These builds include all
+packages intended for the next non-patch Ice release.
+
+## Index
+
+- [Linux]
+  - [Amazon Linux 2023](#amazon-linux-2023)
+  - [Debian 12](#debian-12)
+  - [Red Hat Enterprise Linux 10](#red-hat-enterprise-linux-10)
+  - [Red Hat Enterprise Linux 9](#red-hat-enterprise-linux-9)
+  - [Ubuntu 24.04](#ubuntu-2404)
+- [macOS]
+  - [Swift](#swift)
+  - [Homebrew](#homebrew)
+- [Windows]
+  - [C++ NuGet](#c-nuget)
+  - [MSI Installer](#msi-installer)
+- [.NET 8.0](#net-80)
+- [JavaScript](#javascript)
+- [Java](#java)
+- [MATLAB](#matlab)
+- [Python](#python)
+- [Ruby](#ruby)
+
+---
+
+## Linux
+
+## Amazon Linux 2023
+
+The ZeroC Ice nightly RPM packages for Amazon Linux 2023 are available from the ZeroC DNF nightly repository.
```

Comment:
I find this too verbose.  Maybe convert all these Linux instructions into a single table?